src/Controllers/SchemaController.js

```diff
@@ -1403,15 +1405,11 @@ export default class SchemaController {
     if (perms['requiresAuthentication']) {
       // If aclGroup has * (public)
       if (!aclGroup || aclGroup.length == 0) {
-        throw new Parse.Error(
-          Parse.Error.OBJECT_NOT_FOUND,
-          'Permission denied, user needs to be authenticated.'
-        );
+        const detailedError = 'Permission denied, user needs to be authenticated.';
+        throw createSanitizedError(Parse.Error.OBJECT_NOT_FOUND, detailedError, defaultLogger);
```

Comment:
That's a good concept; could you make this more compact and not add a variable if it's only used in the error obj?


```suggestion
        throw createSanitizedError(
          Parse.Error.OBJECT_NOT_FOUND,
          'Permission denied, user needs to be authenticated.',
          defaultLogger,
        );
```

If the `defaultLogger` is used often, then maybe you can use it as default in the `createSanitizedError` method, so you don't need to pass it every time.